wasm/store.go

```diff
@@ -177,6 +178,27 @@ type (
 	FunctionTypeID uint64
 )
 
+// addExport adds and indexes the given export or errs if the name is already exported.
+func (m *ModuleInstance) addExport(name string, e *ExportInstance) error {
```

Comment:
since module knows its own name, now, we can centralize the errors in the module itself (including the name part of the error message), rather than always passing it the name we instantiated it with.

In the future we can also allow someone to instantiate a module with the name that was serialized into the namesection or in the text format, but I didn't do that now.